core/templates/pages/exploration-editor-page/translation-tab/state-translation/state-translation.component.ts

```diff
@@ -523,6 +542,53 @@ export class StateTranslationComponent implements OnInit, OnDestroy {
     }
   }
 
+  tabStatusColorStyleFeedback(tabId: string): object {
+    if (this.isDisabled(tabId)) {
+      return {};
+    }
+
+    const hasFeedbackContent =
+      this.stateAnswerGroups.some(ag => !ag.outcome.feedback.isEmpty()) ||
+      (this.stateDefaultOutcome &&
+        !this.stateDefaultOutcome.feedback.isEmpty());
+
+    if (
+      (!hasFeedbackContent || this.needsTranslation()) &&
+      this.translationTabActiveModeService.isVoiceoverModeActive()
+    ) {
+      return {
+        'border-top-color':
+          this.translationStatusService.NULL_ASSETS_AVAILABLE_COLOR,
+      };
+    }
+
+    return {
+      'border-top-color':
+        this.translationStatusService.getActiveStateComponentStatusColor(tabId),
+    };
+  }
+
+  tabStatusColorStyleContent(tabId: string): object {
+    if (this.isDisabled(tabId)) {
+      return {};
+    }
+
+    if (
+      this.needsTranslation() &&
+      this.translationTabActiveModeService.isVoiceoverModeActive()
+    ) {
+      return {
```

Comment:
Reduce the duplication and branching. Compute the colour first, then have a single return statement that makes use of it.